flexmeasures/cli/data_add.py

```diff
@@ -480,6 +487,79 @@ def add_beliefs(
             print("As a possible workaround, use the --allow-overwrite flag.")
 
 
+@fm_add_data.command("holidays")
+@with_appcontext
+@click.option(
+    "--year",
+    type=click.INT,
+    help="The year for which to look up holidays",
+)
+@click.option(
+    "--country",
+    "countries",
+    type=click.STRING,
+    multiple=True,
+    help="The ISO 3166-1 country/region or ISO 3166-2 sub-region for which to look up holidays (such as US, BR and DE). This argument can be given multiple times.",
+)
+@click.option(
+    "--asset-id",
+    "generic_asset_ids",
+    type=click.INT,
+    multiple=True,
+    help="Add annotations to this asset. Follow up with the asset's ID. This argument can be given multiple times.",
+)
+@click.option(
+    "--account-id",
+    "account_ids",
+    type=click.INT,
+    multiple=True,
+    help="Add annotations to all assets of this account. Follow up with the account's ID. This argument can be given multiple times.",
+)
+def add_holidays(
+    year: int,
+    countries: List[str],
+    generic_asset_ids: List[int],
+    account_ids: List[int],
+):
+    """Add holiday annotations to assets."""
+    calendars = registry.get_calendars(countries)
+    num_holidays = {}
+    asset_query = db.session.query(GenericAsset)
+    if generic_asset_ids:
+        asset_query = asset_query.filter(GenericAsset.id.in_(generic_asset_ids))
+    if account_ids:
+        asset_query = asset_query.filter(GenericAsset.account_id.in_(account_ids))
+    assets = asset_query.all()
+    for country, calendar in calendars.items():
+        _source = get_or_create_source(
+            "workalendar", model=country, source_type="CLI script"
+        )
+        holidays = calendar().holidays(year)
+        for holiday in holidays:
+            start = pd.Timestamp(holiday[0])
+            end = start + pd.offsets.DateOffset(days=1)
+            annotation = Annotation(
+                name=holiday[1],
+                start=start,
+                end=end,
+                source=_source,
+                type="holiday",
+            )
+            db.session.add(annotation)
+            db.session.flush()
+            for asset in assets:
+                relation = GenericAssetAnnotationRelationship(
```

Comment:
Is it not possible to write `asset.annotations.append(annotation)`?

It's shorter and would help us also to save the `db.session.flush()`

Reply:
I'm learning a lot by looking at your many-to-many relationship implementation of account roles. I'll gladly make these changes.